OrbitTest/main_short_lived_threads.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include <chrono>
+#include <iostream>
+#include <mutex>
+#include <random>
+#include <set>
+#include <string>
+#include <thread>
+#include <vector>
+
+std::mutex g_joinable_mutex;
+std::set<std::thread::id> g_joinable_threads;
+
+// Busy wait for ttl_ms.
+void Worker(int ttl_ms) {
+  auto deadline = std::chrono::system_clock::now() + std::chrono::milliseconds(ttl_ms);
+  while (true) {
+    if (std::chrono::system_clock::now() > deadline) {
+      break;
+    }
+  }
+  std::lock_guard lock(g_joinable_mutex);
+  g_joinable_threads.emplace(std::this_thread::get_id());
```

Comment:
Is the g_joinable_threads set really needed? We could simply create all threads and then join them all, no?

Reply:
I'd like to keep _number_of_threads_ threads running all the time to keep the system under stress. So in every round in the main loop I just want to join the threads that already finished their "work".

Reply:
Ah ok ok. And the lifecycle of threads is what you are interested in? A comment on how this program will be used to add more context could be useful.

Reply:
Yes, I'd like to provoke errors in places where we need to keep track of the set of threads in the target process.

Added a few lines of comments.